File: android/framework_old/src/main/java/com/dreampany/framework/data/source/pref/AdPref.java
/*
package com.dreampany.frame.data.source.pref;

import android.content.Context;

import com.dreampany.frame.util.TimeUtil;

import javax.inject.Inject;
import javax.inject.Singleton;

*/
/**
 * Created by Hawladar Roman on 8/29/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 *//*

@Singleton
public class AdPref extends BasePref {

    private static final String BANNER_TIME = "banner_time";
    private static final String INTERSTITIAL_TIME = "interstitial_time";
    private static final String REWARDED_TIME = "rewarded_time";

    @Inject
    AdPref(Context context) {
        super(context);
    }

    @Override
    protected String getPrivateName(Context context) {
        return getClass().getSimpleName();
    }

    public void setBannerTime(long time) {
        setPrivately(BANNER_TIME, time);
    }

    public void setInterstitialTime(long time) {
        setPrivately(INTERSTITIAL_TIME, time);
    }

    public void setRewardedTime(long time) {
        setPrivately(REWARDED_TIME, time);
    }

    public boolean isBannerTimeExpired(long expireTime) {
        updateIfMissing(BANNER_TIME, TimeUtil.currentTime());
        long time = getPrivately(BANNER_TIME, 0L);
        return TimeUtil.isExpired(time, expireTime);
    }

    public boolean isInterstitialTimeExpired(long expireTime) {
        updateIfMissing(INTERSTITIAL_TIME, TimeUtil.currentTime());
        long time = getPrivately(INTERSTITIAL_TIME, 0L);
        return TimeUtil.isExpired(time, expireTime);
    }

    public boolean isRewardedTimeExpired(long expireTime) {
        updateIfMissing(REWARDED_TIME, TimeUtil.currentTime());
        long time = getPrivately(REWARDED_TIME, 0L);
        return TimeUtil.isExpired(time, expireTime);
    }

    private void updateIfMissing(String key, long value) {
        if (!isPrivateAvailable(key)) {
            setPrivately(key, value);
        }
    }
}
*/

File: android/framework_old/src/main/kotlin/com/dreampany/framework/data/source/pref/AdPref.kt
package com.dreampany.framework.data.source.pref

import android.content.Context
import com.dreampany.framework.misc.Constants
import com.dreampany.framework.util.TimeUtil
import com.dreampany.framework.util.TimeUtilKt
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by roman on 2019-08-15
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
@Singleton
class AdPref
@Inject constructor(context: Context) : FramePref(context) {

    override fun getPrivateName(context: Context): String {
        return Constants.Pref.AD
    }

    fun setBannerTime(time: Long) {
        setPrivately(Constants.AdTime.BANNER, time)
    }

    fun setInterstitialTime(time: Long) {
        setPrivately(Constants.AdTime.INTERSTITIAL, time)
    }

    fun setRewardedTime(time: Long) {
        setPrivately(Constants.AdTime.REWARDED, time)
    }

    fun isBannerTimeExpired(expireTime: Long): Boolean {
        updateIfMissing(Constants.AdTime.BANNER, TimeUtil.currentTime())
        val time = getPrivately(Constants.AdTime.BANNER, 0L)
        return TimeUtilKt.isExpired(time, expireTime)
    }

    fun isInterstitialTimeExpired(expireTime: Long): Boolean {
        updateIfMissing(Constants.AdTime.INTERSTITIAL, TimeUtilKt.currentMillis())
        val time = getPrivately(Constants.AdTime.INTERSTITIAL, 0L)
        return TimeUtilKt.isExpired(time, expireTime)
    }

    fun isRewardedTimeExpired(expireTime: Long): Boolean {
        updateIfMissing(Constants.AdTime.REWARDED, TimeUtilKt.currentMillis())
        val time = getPrivately(Constants.AdTime.REWARDED, 0L)
        return TimeUtilKt.isExpired(time, expireTime)
    }

    private fun updateIfMissing(key: String, value: Long) {
        if (!hasPrivate(key)) {
            setPrivately(key, value)
        }
    }
}